src/pdp/server.ts

```diff
@@ -128,16 +128,19 @@ export class PDPServer {
    * @param serviceName - Service name for uploads (defaults to 'public')
    */
   constructor (
-    authHelper: PDPAuthHelper,
+    authHelper: PDPAuthHelper | null,
     apiEndpoint: string,
     retrievalEndpoint: string,
     serviceName: string = 'public'
   ) {
-    if (apiEndpoint.trim() === '') {
-      throw new Error('PDP API endpoint is required')
-    }
-    if (retrievalEndpoint.trim() === '') {
-      throw new Error('PDP retrieval endpoint is required')
+    // Allow empty endpoints only for null auth (ping validation scenarios)
+    if (authHelper !== null) {
+      if (apiEndpoint.trim() === '') {
+        throw new Error('PDP API endpoint is required')
+      }
+      if (retrievalEndpoint.trim() === '') {
+        throw new Error('PDP retrieval endpoint is required')
+      }
     }
```

Comment:
```suggestion
    if (apiEndpoint.trim() === '') {
      throw new Error('PDP API endpoint is required')
    }
    if (retrievalEndpoint.trim() === '') {
      throw new Error('PDP retrieval endpoint is required')
    }
```

I think we can undo this now, no? We should always expect these parameters, it's not really useful without them.